### PR TITLE
jv-lottery

### PR DESCRIPTION
**Pull request by Ppllupold**

### CODE REVIEW

File: src/main/java/core/basesyntax/Lottery.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class Lottery {
+
+    public static Ball getRandomBall() {
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/Lottery.java:7]: [Don’t use static methods in your solution](https://mate-academy.github.io/jv-program-common-mistakes/java-core/oop/lottery#dont-use-static-methods-in-your-solution)

File: src/main/java/core/basesyntax/Application.java [hunk 2]
```diff
@@ -2,6 +2,9 @@
 
 public class Application {
     public static void main(String[] args) {
-        // create three balls using class Lottery and print information about them in console
+
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/Application.java:5]: [Please don’t add redundant empty lines to your code.](https://mate-academy.github.io/jv-program-common-mistakes/java-core/oop/lottery#please-dont-add-redundant-empty-lines-to-your-code)

Check everywhere

File: src/main/java/core/basesyntax/Lottery.java [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+        return new Ball(ColorSupplier.getRandomColor().name().toLowerCase(),
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/Lottery.java:8]: [Think about which variables should be local in the method and which should be class-level](https://mate-academy.github.io/jv-program-common-mistakes/java-core/oop/lottery#think-about-which-variables-should-be-local-in-the-method-and-which-should-be-class-level)

File: src/main/java/core/basesyntax/Lottery.java [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+                new Random().nextInt(101));
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/Lottery.java:9]: [All magic numbers in your code should be constants.](https://mate-academy.github.io/jv-program-common-mistakes/java-core/oop/lottery#all-magic-numbers-in-your-code-should-be-constants)

File: src/main/java/core/basesyntax/Colors.java [hunk 5]
```diff
@@ -0,0 +1,5 @@
+package core.basesyntax;
+
+public enum Colors {
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/Colors.java:3]: read naming convention for enum
```suggestion
public enum Color {
```

File: src/main/java/core/basesyntax/Colors.java [hunk 6]
```diff
@@ -0,0 +1,5 @@
...
+    RED, BLUE, GREEN;
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/Colors.java:4]:
```suggestion
    RED, BLUE, GREEN
```

File: src/main/java/core/basesyntax/Ball.java [hunk 7]
```diff
@@ -0,0 +1,36 @@
+package core.basesyntax;
+
+public class Ball {
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/Ball.java:3]: Don't create methods that you actually don't use

File: src/main/java/core/basesyntax/ColorSupplier.java [hunk 8]
```diff
@@ -1,7 +1,10 @@
 package core.basesyntax;
 
+import java.util.Random;
+
 public class ColorSupplier {
-    public String getRandomColor() {
-        return null;
+    public static Colors getRandomColor() {
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/ColorSupplier.java:6]:
```suggestion
    public Colors getRandomColor() {
```

File: src/main/java/core/basesyntax/ColorSupplier.java [hunk 9]
```diff
@@ -1,7 +1,10 @@
...
+        int randomColor = new Random().nextInt(Colors.values().length);
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/ColorSupplier.java:7]: new Random() class-level variable

File: src/main/java/core/basesyntax/Application.java [hunk 10]
```diff
@@ -2,6 +2,9 @@
...
+        for (int i = 0; i < 10; i++) {
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/Application.java:6]: 10 is a magic number, create constant

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Please, read carefully Common mistakes file before you create PR

File: src/main/java/core/basesyntax/Application.java [hunk 11]
```diff
@@ -1,7 +1,11 @@
 package core.basesyntax;
 
 public class Application {
+    private static final int PRINT_COUNTER = 10;
+
     public static void main(String[] args) {
-        // create three balls using class Lottery and print information about them in console
+        for (int i = 0; i < PRINT_COUNTER; i++) {
+            System.out.println(new Lottery().getRandomBall());
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/Application.java:8]: Don't need to create new Lottery() object in a loop.

File: src/main/java/core/basesyntax/Ball.java [hunk 12]
```diff
@@ -0,0 +1,20 @@
...
+
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/Ball.java:4]:
```suggestion
```

File: src/main/java/core/basesyntax/ColorSupplier.java [hunk 13]
```diff
@@ -1,7 +1,12 @@
 package core.basesyntax;
 
+import java.util.Random;
+
 public class ColorSupplier {
-    public String getRandomColor() {
-        return null;
+    private int randomColor = new Random().nextInt(Color.values().length);
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/ColorSupplier.java:6]:
```suggestion
    private final Random random = new Random();

    public String getRandomColor() {
        int index = random.nextInt(Color.values().length);
        return Color.values()[index].name().toLowerCase();
    }
```

File: src/main/java/core/basesyntax/Lottery.java [hunk 14]
```diff
@@ -0,0 +1,13 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class Lottery {
+    private static final int rand = 101;
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/Lottery.java:6]: Bad naming - rand. For example, MAX_BALL_NUMBER
Also, read constant convention: Constant names are to be written in uppercase with an underscore separating the words.

File: src/main/java/core/basesyntax/Lottery.java [hunk 15]
```diff
@@ -0,0 +1,13 @@
...
+    private Color color = new ColorSupplier().getRandomColor();
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/Lottery.java:7]:
```suggestion
    private final Random random = new Random();
    private final ColorSupplier colorSupplier = new ColorSupplier();
```

File: src/main/java/core/basesyntax/Lottery.java [hunk 16]
```diff
@@ -0,0 +1,13 @@
...
+
+    public Ball getRandomBall() {
+        return new Ball(color.name().toLowerCase(),
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/Lottery.java:10]:
```suggestion
        return new Ball(colorSupplier.getRandomColor(),
```

File: src/main/java/core/basesyntax/Lottery.java [hunk 17]
```diff
@@ -0,0 +1,13 @@
...
+                new Random().nextInt(rand));
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/Lottery.java:11]:
```suggestion
                random.nextInt(MAX_BALL_NUMBER));
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Please, don't resolve comments by yourself

File: src/main/java/core/basesyntax/Lottery.java [hunk 18]
```diff
@@ -0,0 +1,14 @@
...
+import java.util.Random;
+
+public class Lottery {
+    private static final int MAX_BALL_NUMBER = 101;
+    private final Random random = new Random();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Ball getRandomBall() {
+        return new Ball(colorSupplier.getRandomColor(),
+                random.nextInt(101));
```
Comment (okuzan) [inline src/main/java/core/basesyntax/Lottery.java:12]: why don't you use your constant?

File: pom.xml [hunk 19]
```diff
@@ -9,7 +9,7 @@
     <version>1.0-SNAPSHOT</version>
 
     <properties>
-        <jdk.version>17</jdk.version>
+        <jdk.version>11</jdk.version>
```
Comment (okuzan) [inline pom.xml:12]: why?

**Review by okuzan — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 19 above)
Comment (sarakhmen) [reply pom.xml:12]: please revert this version

**Review by sarakhmen — APPROVED** [review]
Good job :)

**Review by sarakhmen — APPROVED** [review]
Good job:)

**Pull request closed without merging**